File: IUT Blagnac/2ème année/Semestre 3/ProgEff/TP1 & 2 Utilisation des dictionnaires/TP2_python_progEff_PHAKEOVILAY_Andrew_1B/test_of_list.py
import unittest
import dictutils


class MyTestCase(unittest.TestCase):

    def test_core(self):
        li = ["lol:mdr", "bye:ciao"]
        di = dictutils.of_pair_list(li)
        expected = {'lol': 'mdr', 'bye': 'ciao'}
        self.assertEqual(expected, di)

    def test_same_values(self):
        li = ["bonjour:hello", "salut:hello", "coucou:hello"]
        di = dictutils.of_pair_list(li)
        expected = {'bonjour': 'hello', 'salut': 'hello', 'coucou': 'hello'}
        self.assertEqual(expected, di)

    def test_too_few_columns(self):
        li = ["lol_mdr"]
        try:
            di = dictutils.of_pair_list(li)
        except ValueError:
            return  # expected error
        self.fail("Expecting ValueError, got: " + str(di))

    def test_too_much_columns(self):
        li = ["lol:mdr:kikou"]
        try:
            di = dictutils.of_pair_list(li)
        except ValueError:
            return  # expected error
        self.fail("Expecting ValueError, got: " + str(di))


if __name__ == '__main__':
    unittest.main()

File: IUT Blagnac/2ème année/Semestre 3/ProgEff/TP1 & 2 Utilisation des dictionnaires/TP1_java_progEff_PHAKEOVILAY_Andrew_1B/tests/TestOfList.java
package tp1.tests;

import static org.junit.Assert.*;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.junit.Test;

import tp1.MapUtils;

public class TestOfList {

	@Test
	public void testCore() {
		List<String> input = Arrays.asList("lol:mdr", "bye:ciao");
		Map<String, String> dict = MapUtils.ofPairList(input);
		Map<String, String> expected = new HashMap<>();
		expected.put("lol", "mdr");
		expected.put("bye", "ciao");
		assertEquals(expected, dict);
	}
	
	@Test
	public void testSameValues() {
		List<String> input = Arrays.asList("bonjour:hello", "salut:hello", "coucou:hello");
		Map<String, String> dict = MapUtils.ofPairList(input);
		Map<String, String> expected = new HashMap<>();
		expected.put("bonjour", "hello");
		expected.put("salut", "hello");
		expected.put("coucou", "hello");
		assertEquals(expected, dict);
	}

}
